tests/drivers/i2s/i2s_api/src/common.c

```diff
@@ -13,30 +13,30 @@ K_MEM_SLAB_DEFINE(rx_mem_slab, BLOCK_SIZE, NUM_RX_BLOCKS, 32);
 K_MEM_SLAB_DEFINE(tx_mem_slab, BLOCK_SIZE, NUM_TX_BLOCKS, 32);
 
 /* The data_l represent a sine wave */
-ZTEST_DMEM int16_t data_l[SAMPLE_NO] = {
+ZTEST_DMEM int32_t data_l[SAMPLE_NO] = {
 	  6392,  12539,  18204,  23169,  27244,  30272,  32137,  32767,  32137,
 	 30272,  27244,  23169,  18204,  12539,   6392,      0,  -6393, -12540,
 	-18205, -23170, -27245, -30273, -32138, -32767, -32138, -30273, -27245,
 	-23170, -18205, -12540,  -6393,     -1,
 };
 
 /* The data_r represent a sine wave with double the frequency of data_l */
-ZTEST_DMEM int16_t data_r[SAMPLE_NO] = {
+ZTEST_DMEM int32_t data_r[SAMPLE_NO] = {
 	 12539,  23169,  30272,  32767,  30272,  23169,  12539,      0, -12540,
 	-23170, -30273, -32767, -30273, -23170, -12540,     -1,  12539,  23169,
 	 30272,  32767,  30272,  23169,  12539,      0, -12540, -23170, -30273,
 	-32767, -30273, -23170, -12540,     -1,
 };
 
-static void fill_buf(int16_t *tx_block, int att)
+static void fill_buf(int32_t *tx_block, int att)
 {
 	for (int i = 0; i < SAMPLE_NO; i++) {
 		tx_block[2 * i] = data_l[i] >> att;
```

Comment:
With this change, buffers are always filled with 32-bit values, even when the word size is configured to 16 bits. Why?

Reply:
I didn't suggest that it would work after just changing the word size. What I'm pointing out is that those buffers of 32-bit values will be treated as containing 16-bit samples when the word size is set to 16 bits. And this is undesirable.

Reply:
Then i would rather need K-config for word size to make separate set of test data.